operatortrace-go/pkg/client/tracing_status_client.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"go.opentelemetry.io/otel/trace"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+)
+
+type tracingStatusClient struct {
+	scheme *runtime.Scheme
+	client.StatusWriter
+	trace.Tracer
+	Logger logr.Logger
+}
+
+var _ client.StatusWriter = (*tracingStatusClient)(nil)
+
+func (tc *tracingClient) Status() client.StatusWriter {
+	return &tracingStatusClient{
+		scheme:       tc.scheme,
+		Logger:       tc.Logger,
+		StatusWriter: tc.Client.Status(),
+		Tracer:       tc.Tracer,
+	}
+}
+
+func (ts *tracingStatusClient) Update(ctx context.Context, obj client.Object, opts ...client.SubResourceUpdateOption) error {
+	gvk, err := apiutil.GVKForObject(obj, ts.scheme)
+	if err != nil {
+		return fmt.Errorf("problem getting the scheme: %w", err)
+	}
+
+	kind := gvk.GroupKind().Kind
+
+	ctx, span := startSpanFromContext(ctx, ts.Logger, ts.Tracer, obj, ts.scheme, fmt.Sprintf("StatusUpdate %s %s", kind, obj.GetName()))
+	defer span.End()
+
+	setConditionMessage("TraceID", span.SpanContext().TraceID().String(), obj, ts.scheme)
+	setConditionMessage("SpanID", span.SpanContext().SpanID().String(), obj, ts.scheme)
+
+	ts.Logger.Info("updating status object", "object", obj.GetName())
+	err = ts.StatusWriter.Update(ctx, obj, opts...)
```

Comment:
```suggestion
	err = ts.Update(ctx, obj, opts...)
```
Not sure if you have to do that or not but it's non-blocking.

Reply:
updated.

Reply:
I tried this and the test failed when its just `ts.Update`. I am leaving this as is for now.